mule-extensions-api/src/main/java/org/mule/runtime/extension/api/security/AuthenticationHandler.java

```diff
@@ -51,7 +52,7 @@ void setAuthentication(List<String> securityProviders, Authentication authentica
   /**
    * @return the {@link Authentication} in the current context
    */
-  Authentication getAuthentication();
+  Optional<Authentication> getAuthentication();
```

Comment:
Http connector and other test extensions use this. If you change this you need to update them as well. Also verify if soapKit and apikit use it as well.

Reply:
Nobody else seems to be using this method